features/270-next-skycli/api.md

```diff
@@ -113,27 +113,144 @@ access skycli instead of using user account.
 
 - cluster controller endpoint
 - user credentials 
 
-### ./.skyclirc
-- current app
-
-## Discussion
+### ./skygear.yaml
 
-- Current `skycli app view-config` and `skycli app update-config` give the
-greatest flexible to user to update the config. But user need to understand the
-tenant config structure. I think this case is similar to v1 cms config file.
-- Current tenant config is not a simple key value mapping, so we didn't have a
-single command to update particular value. User need to export the current config
-by `skycli app view-config`, update, and upload by `skycli app update-config`.
-- `skycli app view-config` will mask the auth db url, it is also not editable by
-`skycli app update-config`.
-- I think we should not show the secret value in the list comment. But user may
-want to obtain the db credentials after enabling service?
+- config version (If we have new config version in future, skycli will read the
+  version and convert the previous config to the current version.)
+- current app
+- cloud functions configuration
 
-## Questions
+#### Example
+```
+config_version: 1
+app: myapp
+cf:
+  function1:
+    type: http-handler
+    path: /hello-world
+    env: node
+    src: js/hellow-world
+    secrets:
+      - DATABASE_URL
+    permission:
+      - name: key_required
+      - name: user_required
+static:
+  ...
+```
 
-- How to support email template?
+## Use cases
+
+- User creates app and uses skycli for app configuration without CF
+
+    ```sh
+    # Create app
+    $ skycli app create
+
+    ? What is your app name? myapp
+    Creating app...
+    Your API endpoint: https://myapp.api.skygear.io
+    Your Client API Key: DSOJSDOSDKOJASNLSLC
+    Your Master API Key: FJOADJOFAJOFJOASDJK
+    Created app successfully!
+
+    To setup app project directory, run:
+        skycli app scaffold
+
+    # Add user to app
+    $ skycli app add-user dev@example.com --app=myapp
+
+    # Update tenant config to enable welcome email
+    $ skycli app update-tenant-config WELCOME_EMAIL --app=myapp
+    ? Edit application tenant config. Press <enter> to launch your preferred editor.
+    # Enter editor mode with existing config
+    ENABLED: false
+    SENDER: no-reply@skygeario.com
+    SENDER_NAME: ''
+    REPLY_TO: ''
+    REPLY_TO_NAME: ''
+    SUBJECT: Welcome!
+    EMAIL_HTML: |-
+      <p>Hello {% if user.name %}{{ user.name }}{% else %}{{ user.email }}{% endif %},</p>
+
+      <p>Welcome to Skygear.</p>
+
+      <p>Thanks.</p>
+    ~
+    ~
+    ~
+
+    # User don't want to specify app for every commands, create app directory by scaffold command
+    $ skycli app scaffold
+    ? You're about to initialize a Skygear project in this directory: /Users/ubuntu/myapp
+    Confirm? (Y/n)
+
+    Fetching the list of your apps...
+    ? Select an app to associate with the directory: (Use arrow keys)
+    > myapp
+      myapp2
+      myapp3
+    (Move up and down to reveal more choices)
+
+    Fetching examples...
```

Comment:
Is it multiple choice? Each app can have multiple js cloud functions, handler etc... right? So if it only choose one that seems a bit misleading to new user?

Reply:
Or just have another command for people to fetch example...?

Reply:
My original thought for this command, we will have a git repo that store the example project code, each examples should be a complete project that is created based on different use case. Some sample can be very simple like `helloworld-js-function`, we can develop more complicated samples in future like `pyramid-py-service`... etc. The command will fetch the master branch of repo to get the sample list, download the selected sample and change the `skygear.yaml` app only. Since every sample is a complete project, so user can only choose once and we are not going the merge the projects.

The folder structure of the sample repo maybe something like this, one folder per sample.
```
- helloworld-js-function/
- pyramid-py-service/
- ... more sample
- samples.json
```

If we think this is a good idea, maybe we can change the name of the first version list...? To avoid misleading.
```
- helloworld-js-function
- basic-js-handler
- expressjs-service
- auth-gear-hooks
```

If we need to support select multiple, the story will be different. The list may not be define by use case. And we may want to have another commands to let user add handler or function after the project created.